File: backend/app/schemas/company.py
from pydantic import BaseModel, Field, HttpUrl


class CompanyDashboardSummary(BaseModel):
    total_jobs: int
    active_jobs: int
    applications: int
    interviews: int
    hired_candidates: int
    company_views: int
    premium_status: str
    monthly_growth: int


class CompanyDashboardResponse(BaseModel):
    company_id: str
    company_name: str
    summary: CompanyDashboardSummary
    activity: list[dict[str, str]]


class CompanyProfilePayload(BaseModel):
    company_name: str = Field(min_length=2, max_length=160)
    industry: str = Field(min_length=2, max_length=100)
    company_size: str = Field(min_length=1, max_length=50)
    website: HttpUrl
    location: str = Field(min_length=2, max_length=160)
    founded_year: int = Field(ge=1800, le=2100)
    description: str = Field(min_length=20, max_length=2000)
    linkedin_url: HttpUrl | None = None
    twitter_url: HttpUrl | None = None
    benefits: list[str] = Field(default_factory=list)
    culture: list[str] = Field(default_factory=list)
    logo_url: str | None = None


class CompanyProfileResponse(CompanyProfilePayload):
    uuid: str


class CompanyJobPayload(BaseModel):
    title: str = Field(min_length=2, max_length=160)
    department: str = Field(min_length=2, max_length=100)
    employment_type: str
    experience_level: str
    salary_range: str
    location: str
    work_mode: str
    skills: list[str] = Field(min_length=1)
    education: str
    description: str = Field(min_length=20)
    responsibilities: list[str] = Field(min_length=1)
    requirements: list[str] = Field(min_length=1)
    benefits: list[str] = Field(default_factory=list)
    application_deadline: str
    status: str = "draft"


class CompanyJobResponse(CompanyJobPayload):
    id: str
    applications: int = 0
    views: int = 0
    created_at: str


class CompanyJobListResponse(BaseModel):
    jobs: list[CompanyJobResponse]
    total: int


class CompanyApplicant(BaseModel):
    id: str
    name: str
    role: str
    experience: str
    location: str
    match_score: int = Field(ge=0, le=100)
    status: str
    applied_at: str
    skills: list[str]


class CompanyApplicantListResponse(BaseModel):
    applicants: list[CompanyApplicant]
    total: int


class CompanyAnalyticsResponse(BaseModel):
    applications_trend: list[dict[str, int | str]]
    hiring_funnel: list[dict[str, int | str]]
    job_performance: list[dict[str, int | str]]

File: frontend/src/types/company.ts
import type { IconType } from 'react-icons';

export interface CompanyNavItem {
  id: string;
  label: string;
  to: string;
  icon: IconType;
  end?: boolean;
}

export interface CompanyDashboardSummary {
  total_jobs: number;
  active_jobs: number;
  applications: number;
  interviews: number;
  hired_candidates: number;
  company_views: number;
  premium_status: string;
  monthly_growth: number;
}

export interface CompanyDashboardResponse {
  company_id: string;
  company_name: string;
  summary: CompanyDashboardSummary;
  activity: Array<{ title: string; time: string }>;
}

export interface CompanyProfile {
  uuid?: string;
  company_name: string;
  industry: string;
  company_size: string;
  website: string;
  location: string;
  founded_year: number;
  description: string;
  linkedin_url?: string | null;
  twitter_url?: string | null;
  benefits: string[];
  culture: string[];
  logo_url?: string | null;
}

export interface CompanyJob {
  id?: string;
  title: string;
  company: string;
  department: string;
  employment_type: string;
  experience_level: string;
  salary_range: string;
  location: string;
  work_mode: string;
  skills: string[];
  preferred_skills: string[];
  education: string;
  description: string;
  responsibilities: string[];
  requirements: string[];
  benefits: string[];
  openings: number;
  category: string;
  tags: string[];
  application_deadline: string;
  status: CompanyJobStatus;
  applications?: number;
  views?: number;
  created_at?: string;
}

export type CompanyJobStatus = 'archived' | 'closed' | 'draft' | 'published';

export interface CompanyJobAnalytics {
  job_id: string;
  views: number;
  applications: number;
  shortlisted: number;
  interviews: number;
  conversion_rate: number;
  trend: Array<{ label: string; views: number; applications: number }>;
}

export interface CompanyApplicant {
  id: string;
  name: string;
  role: string;
  experience: string;
  location: string;
  match_score: number;
  status: string;
  applied_at: string;
  skills: string[];
}

export interface CompanyAnalytics {
  applications_trend: Array<{ label: string; applications: number; views: number }>;
  hiring_funnel: Array<{ stage: string; value: number }>;
  job_performance: Array<{ name: string; views: number; applications: number }>;
}
